// @angular/material/form-field/typings/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/material/form-field';
import * as i1 from '@angular/common';
import * as i2 from '@angular/core';
import * as i3 from '@angular/cdk/platform';
import * as i4 from '../../../common/common.ngsummary';
import * as i5 from '../../../cdk/platform/typings/index.ngsummary';
import * as i6 from '@angular/material/core';
import * as i7 from './index.ngfactory';
export function MatFormFieldModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MatFormFieldModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i1.NgLocalization}},
      useClass:{reference:i1.NgLocaleLocalization,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
      useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
          isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
      multi:false},module:{reference:i1.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i3.Platform,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          useClass:{reference:i3.Platform,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
          deps:([] as any[]),multi:false},module:{reference:i3.PlatformModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}}],modules:[{reference:i1.CommonModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},{reference:i3.PlatformModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},{reference:i0.MatFormFieldModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])}],exportedDirectives:[{reference:i0.MatError},{reference:i0.MatHint},
      {reference:i0.MatFormField},{reference:i0.MatPlaceholder},{reference:i0.MatPrefix},
      {reference:i0.MatSuffix}],exportedPipes:([] as any[])},MatErrorNgSummary,MatHintNgSummary,
      MatFormFieldNgSummary,MatPlaceholderNgSummary,MatPrefixNgSummary,MatSuffixNgSummary,
      i4.CommonModuleNgSummary,i5.PlatformModuleNgSummary];
}
export function MatErrorNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatError,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'mat-error',exportAs:(null as any),inputs:{id:'id'},
      outputs:{},hostListeners:{},hostProperties:{'attr.id':'id'},hostAttributes:{class:'mat-error',
          role:'alert'},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function MatHintNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatHint,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'mat-hint',exportAs:(null as any),inputs:{align:'align',
          id:'id'},outputs:{},hostListeners:{},hostProperties:{'class.mat-right':'align == "end"',
          'attr.id':'id','attr.align':'null'},hostAttributes:{class:'mat-hint'},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatFormFieldNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatFormField,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.Renderer2}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ChangeDetectorRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i6.MAT_PLACEHOLDER_GLOBAL_OPTIONS}}}],lifecycleHooks:[4,
      5,6]},isComponent:true,selector:'mat-input-container, mat-form-field',exportAs:'matFormField',
      inputs:{color:'color',dividerColor:'dividerColor',hideRequiredMarker:'hideRequiredMarker',
          hintLabel:'hintLabel',floatPlaceholder:'floatPlaceholder'},outputs:{},hostListeners:{},
      hostProperties:{'class.mat-input-invalid':'_control.errorState','class.mat-form-field-invalid':'_control.errorState',
          'class.mat-form-field-can-float':'_canPlaceholderFloat','class.mat-form-field-should-float':'_control.shouldPlaceholderFloat || _shouldAlwaysFloat',
          'class.mat-focused':'_control.focused','class.mat-primary':'color == "primary"',
          'class.mat-accent':'color == "accent"','class.mat-warn':'color == "warn"',
          'class.ng-untouched':'_shouldForward("untouched")','class.ng-touched':'_shouldForward("touched")',
          'class.ng-pristine':'_shouldForward("pristine")','class.ng-dirty':'_shouldForward("dirty")',
          'class.ng-valid':'_shouldForward("valid")','class.ng-invalid':'_shouldForward("invalid")',
          'class.ng-pending':'_shouldForward("pending")'},hostAttributes:{class:'mat-input-container mat-form-field'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:[{selectors:[{identifier:{reference:i0.MatFormFieldControl}}],
          first:true,descendants:true,propertyName:'_control',read:(null as any)},
          {selectors:[{identifier:{reference:i0.MatPlaceholder}}],first:true,descendants:true,
              propertyName:'_placeholderChild',read:(null as any)},{selectors:[{identifier:{reference:i0.MatError}}],
              first:false,descendants:false,propertyName:'_errorChildren',read:(null as any)},
          {selectors:[{identifier:{reference:i0.MatHint}}],first:false,descendants:false,
              propertyName:'_hintChildren',read:(null as any)},{selectors:[{identifier:{reference:i0.MatPrefix}}],
              first:false,descendants:false,propertyName:'_prefixChildren',read:(null as any)},
          {selectors:[{identifier:{reference:i0.MatSuffix}}],first:false,descendants:false,
              propertyName:'_suffixChildren',read:(null as any)}],viewQueries:[{selectors:[{value:'underline'}],
          first:true,descendants:true,propertyName:'underlineRef',read:(null as any)},
          {selectors:[{value:'connectionContainer'}],first:true,descendants:true,propertyName:'_connectionContainerRef',
              read:(null as any)},{selectors:[{value:'placeholder'}],first:true,descendants:true,
              propertyName:'_placeholder',read:(null as any)}],entryComponents:([] as any[]),
      changeDetection:0,template:{animations:['transitionMessages'],ngContentSelectors:['[matPrefix]',
          '*','mat-placeholder','[matSuffix]','mat-error','mat-hint:not([align=\'end\'])',
          'mat-hint[align=\'end\']'],encapsulation:2},componentViewType:i7.View_MatFormField_0,
      rendererType:i7.RenderType_MatFormField,componentFactory:i7.MatFormFieldNgFactory}];
}
export function MatPlaceholderNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatPlaceholder,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'mat-placeholder',exportAs:(null as any),inputs:{},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatPrefixNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatPrefix,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'[matPrefix]',exportAs:(null as any),inputs:{},outputs:{},
      hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatSuffixNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatSuffix,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'[matSuffix]',exportAs:(null as any),inputs:{},outputs:{},
      hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
