// @ngrx/store/store.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@ngrx/store';
export function StoreModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.StoreModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:([] as any[]),modules:[{reference:i0.StoreModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}],exportedDirectives:([] as any[]),
      exportedPipes:([] as any[])}];
}
export function StoreRootModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.StoreRootModule,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ActionsSubject}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.ReducerObservable}}},{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ScannedActionsSubject}}}],
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:([] as any[]),
      modules:[{reference:i0.StoreRootModule,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ActionsSubject}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i0.ReducerObservable}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ScannedActionsSubject}}}],
          lifecycleHooks:([] as any[])}],exportedDirectives:([] as any[]),exportedPipes:([] as any[])}];
}
export function StoreFeatureModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.StoreFeatureModule,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.STORE_FEATURES}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.FEATURE_REDUCERS}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ReducerManager}}}],
      lifecycleHooks:[1]},entryComponents:([] as any[]),providers:([] as any[]),modules:[{reference:i0.StoreFeatureModule,
      diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.STORE_FEATURES}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.FEATURE_REDUCERS}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i0.ReducerManager}}}],lifecycleHooks:[1]}],
      exportedDirectives:([] as any[]),exportedPipes:([] as any[])}];
}
export function StoreNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.Store,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.StateObservable}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.ActionsSubject}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ReducerManager}}}],
      lifecycleHooks:([] as any[])}}];
}
export function ActionsSubjectNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.ActionsSubject,diDeps:([] as any[]),lifecycleHooks:[1]}}];
}
export function ReducerManagerNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.ReducerManager,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ReducerManagerDispatcher}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.INITIAL_STATE}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.INITIAL_REDUCERS}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.REDUCER_FACTORY}}}],lifecycleHooks:[1]}}];
}
export function ReducerManagerDispatcherNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.ReducerManagerDispatcher,diDeps:([] as any[]),
      lifecycleHooks:[1]}}];
}
export function ScannedActionsSubjectNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.ScannedActionsSubject,diDeps:([] as any[]),
      lifecycleHooks:[1]}}];
}
export function StateNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.State,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ActionsSubject}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.ReducerObservable}}},{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ScannedActionsSubject}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.INITIAL_STATE}}}],lifecycleHooks:[1]}}];
}
