// @angular/material/core/typings/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/material/core';
import * as i1 from '@angular/cdk/bidi';
import * as i2 from '@angular/platform-browser';
import * as i3 from '../../../cdk/bidi/typings/index.ngsummary';
import * as i4 from '@angular/core';
import * as i5 from '@angular/cdk/platform';
import * as i6 from '@angular/common';
import * as i7 from '../../../cdk/platform/typings/index.ngsummary';
import * as i8 from '../../../common/common.ngsummary';
import * as i9 from './index.ngfactory';
export function MatCommonModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MatCommonModule,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}}}],
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i1.DIR_DOCUMENT}},
      useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
      useExisting:{identifier:{reference:i2.DOCUMENT}},deps:(undefined as any),multi:false},
      module:{reference:i1.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i1.Directionality,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
          lifecycleHooks:([] as any[])}},useClass:{reference:i1.Directionality,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
          multi:false},module:{reference:i1.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}},useClass:(null as any),
          useValue:true,useFactory:(null as any),useExisting:(undefined as any),deps:(undefined as any),
          multi:false},module:{reference:i0.MatCommonModule,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}}}],
          lifecycleHooks:([] as any[])}}],modules:[{reference:i0.CompatibilityModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i1.BidiModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i0.MatCommonModule,
      diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}}}],lifecycleHooks:([] as any[])}],
      exportedDirectives:[{reference:i0.MatPrefixRejector},{reference:i0.MdPrefixRejector},
          {reference:i1.Dir}],exportedPipes:([] as any[])},CompatibilityModuleNgSummary,
      i3.BidiModuleNgSummary];
}
export function CompatibilityModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.CompatibilityModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:([] as any[]),
      modules:[{reference:i0.CompatibilityModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}],
      exportedDirectives:[{reference:i0.MatPrefixRejector},{reference:i0.MdPrefixRejector}],
      exportedPipes:([] as any[])},MatPrefixRejectorNgSummary,MdPrefixRejectorNgSummary];
}
export function NoConflictStyleCompatibilityModeNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.NoConflictStyleCompatibilityMode,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:([] as any[]),
      modules:[{reference:i0.NoConflictStyleCompatibilityMode,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}],exportedDirectives:([] as any[]),exportedPipes:([] as any[])}];
}
export function NativeDateModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.NativeDateModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i0.DateAdapter}},
      useClass:{reference:i0.NativeDateAdapter,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MAT_DATE_LOCALE}}}],
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
      useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
          isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MAT_DATE_LOCALE}}}],
      multi:false},module:{reference:i0.NativeDateModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i0.MAT_DATE_LOCALE}},useClass:(null as any),
          useValue:(undefined as any),useFactory:(null as any),useExisting:{identifier:{reference:i4.LOCALE_ID}},
          deps:(undefined as any),multi:false},module:{reference:i0.NativeDateModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],modules:[{reference:i0.NativeDateModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])}],exportedDirectives:([] as any[]),
      exportedPipes:([] as any[])},{summaryKind:3,type:{reference:i0.NativeDateAdapter,
      diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i0.MAT_DATE_LOCALE}}}],lifecycleHooks:([] as any[])}}];
}
export function MatNativeDateModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MatNativeDateModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i0.DateAdapter}},
      useClass:{reference:i0.NativeDateAdapter,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MAT_DATE_LOCALE}}}],
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
      useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
          isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MAT_DATE_LOCALE}}}],
      multi:false},module:{reference:i0.NativeDateModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i0.MAT_DATE_LOCALE}},useClass:(null as any),
          useValue:(undefined as any),useFactory:(null as any),useExisting:{identifier:{reference:i4.LOCALE_ID}},
          deps:(undefined as any),multi:false},module:{reference:i0.NativeDateModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.MAT_DATE_FORMATS}},
          useClass:(null as any),useValue:i0.MAT_NATIVE_DATE_FORMATS,useFactory:(null as any),
          useExisting:(undefined as any),deps:(undefined as any),multi:false},module:{reference:i0.MatNativeDateModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],modules:[{reference:i0.NativeDateModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i0.MatNativeDateModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])}],exportedDirectives:([] as any[]),
      exportedPipes:([] as any[])},NativeDateModuleNgSummary];
}
export function MatLineModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MatLineModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i1.DIR_DOCUMENT}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i2.DOCUMENT}},deps:(undefined as any),
          multi:false},module:{reference:i1.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i1.Directionality,diDeps:[{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])}},useClass:{reference:i1.Directionality,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
              multi:false},module:{reference:i1.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}},useClass:(null as any),
              useValue:true,useFactory:(null as any),useExisting:(undefined as any),
              deps:(undefined as any),multi:false},module:{reference:i0.MatCommonModule,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}}}],
              lifecycleHooks:([] as any[])}}],modules:[{reference:i0.CompatibilityModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i1.BidiModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i0.MatCommonModule,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
              token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}}}],lifecycleHooks:([] as any[])},
          {reference:i0.MatLineModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}],
      exportedDirectives:[{reference:i0.MatPrefixRejector},{reference:i0.MdPrefixRejector},
          {reference:i1.Dir},{reference:i0.MatLine}],exportedPipes:([] as any[])},
      MatLineNgSummary,CompatibilityModuleNgSummary,i3.BidiModuleNgSummary,MatCommonModuleNgSummary];
}
export function MatOptionModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MatOptionModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i1.DIR_DOCUMENT}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i2.DOCUMENT}},deps:(undefined as any),
          multi:false},module:{reference:i1.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i1.Directionality,diDeps:[{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])}},useClass:{reference:i1.Directionality,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
              multi:false},module:{reference:i1.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}},useClass:(null as any),
              useValue:true,useFactory:(null as any),useExisting:(undefined as any),
              deps:(undefined as any),multi:false},module:{reference:i0.MatCommonModule,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}}}],
              lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i5.Platform,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},useClass:{reference:i5.Platform,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])},useValue:(undefined as any),
              useFactory:(null as any),useExisting:(undefined as any),deps:([] as any[]),
              multi:false},module:{reference:i5.PlatformModule,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i6.NgLocalization}},
              useClass:{reference:i6.NgLocaleLocalization,diDeps:[{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i4.LOCALE_ID}}}],
                  lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i4.LOCALE_ID}}}],
              multi:false},module:{reference:i6.CommonModule,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])}}],modules:[{reference:i0.CompatibilityModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i1.BidiModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i0.MatCommonModule,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
              token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}}}],lifecycleHooks:([] as any[])},
          {reference:i5.PlatformModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i0.MatRippleModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i6.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i0.MatPseudoCheckboxModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i0.MatOptionModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}],
      exportedDirectives:[{reference:i0.MatOption},{reference:i0.MatOptgroup}],exportedPipes:([] as any[])},
      MatOptionNgSummary,MatOptgroupNgSummary,CompatibilityModuleNgSummary,i3.BidiModuleNgSummary,
      MatCommonModuleNgSummary,i7.PlatformModuleNgSummary,MatRippleModuleNgSummary,
      i8.CommonModuleNgSummary,MatPseudoCheckboxModuleNgSummary];
}
export function MatRippleModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MatRippleModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i1.DIR_DOCUMENT}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i2.DOCUMENT}},deps:(undefined as any),
          multi:false},module:{reference:i1.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i1.Directionality,diDeps:[{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])}},useClass:{reference:i1.Directionality,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
              multi:false},module:{reference:i1.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}},useClass:(null as any),
              useValue:true,useFactory:(null as any),useExisting:(undefined as any),
              deps:(undefined as any),multi:false},module:{reference:i0.MatCommonModule,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}}}],
              lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i5.Platform,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},useClass:{reference:i5.Platform,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])},useValue:(undefined as any),
              useFactory:(null as any),useExisting:(undefined as any),deps:([] as any[]),
              multi:false},module:{reference:i5.PlatformModule,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])}}],modules:[{reference:i0.CompatibilityModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i1.BidiModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i0.MatCommonModule,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
              token:{identifier:{reference:i0.MATERIAL_SANITY_CHECKS}}}],lifecycleHooks:([] as any[])},
          {reference:i5.PlatformModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i0.MatRippleModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}],
      exportedDirectives:[{reference:i0.MatPrefixRejector},{reference:i0.MdPrefixRejector},
          {reference:i1.Dir},{reference:i0.MatRipple}],exportedPipes:([] as any[])},
      MatRippleNgSummary,CompatibilityModuleNgSummary,i3.BidiModuleNgSummary,MatCommonModuleNgSummary,
      i7.PlatformModuleNgSummary];
}
export function MatPseudoCheckboxModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MatPseudoCheckboxModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:([] as any[]),
      modules:[{reference:i0.MatPseudoCheckboxModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}],
      exportedDirectives:[{reference:i0.MatPseudoCheckbox}],exportedPipes:([] as any[])},
      MatPseudoCheckboxNgSummary];
}
export function MatPrefixRejectorNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatPrefixRejector,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:false,selector:'\n  [mat-button],\n  [mat-fab],\n  [mat-icon-button],\n  [mat-mini-fab],\n  [mat-raised-button],\n  [matCardSubtitle],\n  [matCardTitle],\n  [matCellDef],\n  [matColumnDef],\n  [matDialogActions],\n  [matDialogClose],\n  [matDialogContent],\n  [matDialogTitle],\n  [matHeaderCellDef],\n  [matHeaderRowDef],\n  [matLine],\n  [matRowDef],\n  [matStepLabel],\n  [matStepperNext],\n  [matStepperPrevious],\n  [matTabLabel],\n  [matTabLink],\n  [matTabNav],\n  [matTooltip],\n  [matInput],\n  [matPrefix],\n  [matSuffix],\n  mat-autocomplete,\n  mat-button-toggle,\n  mat-button-toggle,\n  mat-button-toggle-group,\n  mat-card,\n  mat-card-actions,\n  mat-card-content,\n  mat-card-footer,\n  mat-card-header,\n  mat-card-subtitle,\n  mat-card-title,\n  mat-card-title-group,\n  mat-cell,\n  mat-checkbox,\n  mat-chip,\n  mat-dialog-actions,\n  mat-dialog-container,\n  mat-dialog-content,\n  mat-divider,\n  mat-error,\n  mat-grid-list,\n  mat-grid-tile,\n  mat-grid-tile-footer,\n  mat-grid-tile-header,\n  mat-header-cell,\n  mat-header-row,\n  mat-hint,\n  mat-horizontal-stepper,\n  mat-icon,\n  mat-input-container,\n  mat-form-field,\n  mat-list,\n  mat-list-item,\n  mat-menu,\n  mat-nav-list,\n  mat-option,\n  mat-placeholder,\n  mat-progress-bar,\n  mat-pseudo-checkbox,\n  mat-radio-button,\n  mat-radio-group,\n  mat-row,\n  mat-select,\n  mat-sidenav,\n  mat-sidenav-container,\n  mat-slider,\n  mat-spinner,\n  mat-step,\n  mat-tab,\n  mat-table,\n  mat-tab-group,\n  mat-toolbar,\n  mat-vertical-stepper',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function MdPrefixRejectorNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MdPrefixRejector,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:false,selector:'\n  [md-button],\n  [md-fab],\n  [md-icon-button],\n  [md-mini-fab],\n  [md-raised-button],\n  [mdCardSubtitle],\n  [mdCardTitle],\n  [mdCellDef],\n  [mdColumnDef],\n  [mdDialogActions],\n  [mdDialogClose],\n  [mdDialogContent],\n  [mdDialogTitle],\n  [mdHeaderCellDef],\n  [mdHeaderRowDef],\n  [mdLine],\n  [mdRowDef],\n  [mdStepLabel],\n  [mdStepperNext],\n  [mdStepperPrevious],\n  [mdTabLabel],\n  [mdTabLink],\n  [mdTabNav],\n  [mdTooltip],\n  [mdInput],\n  [mdPrefix],\n  [mdSuffix],\n  md-autocomplete,\n  md-button-toggle,\n  md-button-toggle,\n  md-button-toggle-group,\n  md-card,\n  md-card-actions,\n  md-card-content,\n  md-card-footer,\n  md-card-header,\n  md-card-subtitle,\n  md-card-title,\n  md-card-title-group,\n  md-cell,\n  md-checkbox,\n  md-chip,\n  md-dialog-actions,\n  md-dialog-container,\n  md-dialog-content,\n  md-divider,\n  md-error,\n  md-grid-list,\n  md-grid-tile,\n  md-grid-tile-footer,\n  md-grid-tile-header,\n  md-header-cell,\n  md-header-row,\n  md-hint,\n  md-horizontal-stepper,\n  md-icon,\n  md-input-container,\n  md-form-field,\n  md-list,\n  md-list-item,\n  md-menu,\n  md-nav-list,\n  md-option,\n  md-placeholder,\n  md-progress-bar,\n  md-pseudo-checkbox,\n  md-radio-button,\n  md-radio-group,\n  md-row,\n  md-select,\n  md-sidenav,\n  md-sidenav-container,\n  md-slider,\n  md-spinner,\n  md-step,\n  md-tab,\n  md-table,\n  md-tab-group,\n  md-toolbar,\n  md-vertical-stepper',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function MatLineNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatLine,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'[mat-line], [matLine]',exportAs:(null as any),inputs:{},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-line'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatOptionNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatOption,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i4.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i4.ChangeDetectorRef}}},{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MatOptgroup}}}],
      lifecycleHooks:([] as any[])},isComponent:true,selector:'mat-option',exportAs:'matOption',
      inputs:{value:'value',disabled:'disabled'},outputs:{onSelectionChange:'onSelectionChange'},
      hostListeners:{click:'_selectViaInteraction()',keydown:'_handleKeydown($event)'},
      hostProperties:{'attr.tabindex':'_getTabIndex()','class.mat-selected':'selected',
          'class.mat-option-multiple':'multiple','class.mat-active':'active',id:'id',
          'attr.aria-selected':'selected.toString()','attr.aria-disabled':'disabled.toString()',
          'class.mat-option-disabled':'disabled'},hostAttributes:{role:'option',class:'mat-option'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:['*'],encapsulation:2},componentViewType:i9.View_MatOption_0,
      rendererType:i9.RenderType_MatOption,componentFactory:i9.MatOptionNgFactory}];
}
export function MatOptgroupNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatOptgroup,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:true,selector:'mat-optgroup',exportAs:'matOptgroup',inputs:{disabled:'disabled',
          label:'label'},outputs:{},hostListeners:{},hostProperties:{'class.mat-optgroup-disabled':'disabled',
          'attr.aria-disabled':'disabled.toString()','attr.aria-labelledby':'_labelId'},
      hostAttributes:{class:'mat-optgroup',role:'group'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:0,template:{animations:([] as any[]),ngContentSelectors:['mat-option'],
          encapsulation:2},componentViewType:i9.View_MatOptgroup_0,rendererType:i9.RenderType_MatOptgroup,
      componentFactory:i9.MatOptgroupNgFactory}];
}
export function MatRippleNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatRipple,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i4.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i4.NgZone}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.Platform}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i0.MAT_RIPPLE_GLOBAL_OPTIONS}}}],lifecycleHooks:[1,
      3]},isComponent:false,selector:'[mat-ripple], [matRipple]',exportAs:'matRipple',
      inputs:{trigger:'matRippleTrigger',centered:'matRippleCentered',disabled:'matRippleDisabled',
          radius:'matRippleRadius',speedFactor:'matRippleSpeedFactor',color:'matRippleColor',
          unbounded:'matRippleUnbounded'},outputs:{},hostListeners:{},hostProperties:{'class.mat-ripple-unbounded':'unbounded'},
      hostAttributes:{class:'mat-ripple'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatPseudoCheckboxNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatPseudoCheckbox,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:true,selector:'mat-pseudo-checkbox',
      exportAs:(null as any),inputs:{state:'state',disabled:'disabled'},outputs:{},
      hostListeners:{},hostProperties:{'class.mat-pseudo-checkbox-indeterminate':'state === "indeterminate"',
          'class.mat-pseudo-checkbox-checked':'state === "checked"','class.mat-pseudo-checkbox-disabled':'disabled'},
      hostAttributes:{class:'mat-pseudo-checkbox'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:0,template:{animations:([] as any[]),ngContentSelectors:([] as any[]),
          encapsulation:2},componentViewType:i9.View_MatPseudoCheckbox_0,rendererType:i9.RenderType_MatPseudoCheckbox,
      componentFactory:i9.MatPseudoCheckboxNgFactory}];
}
export function NativeDateAdapterNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.NativeDateAdapter,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MAT_DATE_LOCALE}}}],
      lifecycleHooks:([] as any[])}}];
}
export function ShowOnDirtyErrorStateMatcherNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.ShowOnDirtyErrorStateMatcher,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])}}];
}
export function ErrorStateMatcherNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.ErrorStateMatcher,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])}}];
}
export function GestureConfigNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.GestureConfig,diDeps:([] as any[]),lifecycleHooks:([] as any[])}}];
}
