// @angular/common/http/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/common/http';
import * as i1 from '@angular/common';
import * as i2 from '@angular/core';
export function HttpClientJsonpModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.HttpClientJsonpModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i0.JsonpClientBackend,
      diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.ɵb}}},{isAttribute:false,isHost:false,isSelf:false,
          isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.DOCUMENT}}}],
      lifecycleHooks:([] as any[])}},useClass:{reference:i0.JsonpClientBackend,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ɵb}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i1.DOCUMENT}}}],lifecycleHooks:([] as any[])},
      useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
      deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.ɵb}}},{isAttribute:false,isHost:false,isSelf:false,
          isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.DOCUMENT}}}],
      multi:false},module:{reference:i0.HttpClientJsonpModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.ɵb}},
      useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i0.ɵc,
          diDeps:([] as any[])},useExisting:(undefined as any),deps:([] as any[]),
      multi:false},module:{reference:i0.HttpClientJsonpModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.HTTP_INTERCEPTORS}},
      useClass:{reference:i0.JsonpInterceptor,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.JsonpClientBackend}}}],
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
      useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
          isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.JsonpClientBackend}}}],
      multi:true},module:{reference:i0.HttpClientJsonpModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])}}],modules:[{reference:i0.HttpClientJsonpModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])}],exportedDirectives:([] as any[]),
      exportedPipes:([] as any[])},{summaryKind:3,type:{reference:i0.JsonpClientBackend,
      diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.ɵb}}},{isAttribute:false,isHost:false,isSelf:false,
          isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.DOCUMENT}}}],
      lifecycleHooks:([] as any[])}},{summaryKind:3,type:{reference:i0.JsonpInterceptor,
      diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.JsonpClientBackend}}}],lifecycleHooks:([] as any[])}}];
}
export function HttpClientModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.HttpClientModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i0.ɵh,
      diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.HttpXsrfTokenExtractor}}},{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ɵf}}}],
      lifecycleHooks:([] as any[])}},useClass:{reference:i0.ɵh,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.HttpXsrfTokenExtractor}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.ɵf}}}],lifecycleHooks:([] as any[])},useValue:(undefined as any),
      useFactory:(null as any),useExisting:(undefined as any),deps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.HttpXsrfTokenExtractor}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i0.ɵf}}}],multi:false},module:{reference:i0.HttpClientXsrfModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.HTTP_INTERCEPTORS}},
      useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
      useExisting:{identifier:{reference:i0.ɵh}},deps:(undefined as any),multi:true},
      module:{reference:i0.HttpClientXsrfModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i0.HttpXsrfTokenExtractor}},useClass:{reference:i0.ɵg,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i1.DOCUMENT}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.PLATFORM_ID}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i0.ɵe}}}],lifecycleHooks:([] as any[])},
          useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i1.DOCUMENT}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.PLATFORM_ID}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i0.ɵe}}}],multi:false},module:{reference:i0.HttpClientXsrfModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.ɵe}},
          useClass:(null as any),useValue:'XSRF-TOKEN',useFactory:(null as any),useExisting:(undefined as any),
          deps:(undefined as any),multi:false},module:{reference:i0.HttpClientXsrfModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.ɵf}},
          useClass:(null as any),useValue:'X-XSRF-TOKEN',useFactory:(null as any),
          useExisting:(undefined as any),deps:(undefined as any),multi:false},module:{reference:i0.HttpClientXsrfModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.ɵe}},
          useClass:(null as any),useValue:'XSRF-TOKEN',useFactory:(null as any),useExisting:(undefined as any),
          deps:(undefined as any),multi:false},module:{reference:i0.HttpClientModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.ɵf}},
          useClass:(null as any),useValue:'X-XSRF-TOKEN',useFactory:(null as any),
          useExisting:(undefined as any),deps:(undefined as any),multi:false},module:{reference:i0.HttpClientModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.HttpClient,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i0.HttpHandler}}}],lifecycleHooks:([] as any[])}},
          useClass:{reference:i0.HttpClient,diDeps:[{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.HttpHandler}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.HttpHandler}}}],
          multi:false},module:{reference:i0.HttpClientModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.HttpHandler}},
          useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i0.ɵinterceptingHandler,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i0.HttpBackend}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
                      token:{identifier:{reference:i0.HTTP_INTERCEPTORS}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i0.HttpBackend}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.HTTP_INTERCEPTORS}}}],
          multi:false},module:{reference:i0.HttpClientModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.HttpXhrBackend,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i0.XhrFactory}}}],lifecycleHooks:([] as any[])}},
          useClass:{reference:i0.HttpXhrBackend,diDeps:[{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.XhrFactory}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.XhrFactory}}}],
          multi:false},module:{reference:i0.HttpClientModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.HttpBackend}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i0.HttpXhrBackend}},deps:(undefined as any),
          multi:false},module:{reference:i0.HttpClientModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.ɵd,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},useClass:{reference:i0.ɵd,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},useValue:(undefined as any),
          useFactory:(null as any),useExisting:(undefined as any),deps:([] as any[]),
          multi:false},module:{reference:i0.HttpClientModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.XhrFactory}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i0.ɵd}},deps:(undefined as any),multi:false},
          module:{reference:i0.HttpClientModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],
      modules:[{reference:i0.HttpClientXsrfModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i0.HttpClientModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}],
      exportedDirectives:([] as any[]),exportedPipes:([] as any[])},HttpClientXsrfModuleNgSummary,
      {summaryKind:3,type:{reference:i0.HttpClient,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.HttpHandler}}}],
          lifecycleHooks:([] as any[])}},{summaryKind:3,type:{reference:i0.HttpXhrBackend,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i0.XhrFactory}}}],lifecycleHooks:([] as any[])}},
      {summaryKind:3,type:{reference:i0.ɵd,diDeps:([] as any[]),lifecycleHooks:([] as any[])}}];
}
export function HttpClientXsrfModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.HttpClientXsrfModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i0.ɵh,
      diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.HttpXsrfTokenExtractor}}},{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ɵf}}}],
      lifecycleHooks:([] as any[])}},useClass:{reference:i0.ɵh,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.HttpXsrfTokenExtractor}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.ɵf}}}],lifecycleHooks:([] as any[])},useValue:(undefined as any),
      useFactory:(null as any),useExisting:(undefined as any),deps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.HttpXsrfTokenExtractor}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i0.ɵf}}}],multi:false},module:{reference:i0.HttpClientXsrfModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.HTTP_INTERCEPTORS}},
      useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
      useExisting:{identifier:{reference:i0.ɵh}},deps:(undefined as any),multi:true},
      module:{reference:i0.HttpClientXsrfModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i0.HttpXsrfTokenExtractor}},useClass:{reference:i0.ɵg,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i1.DOCUMENT}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.PLATFORM_ID}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i0.ɵe}}}],lifecycleHooks:([] as any[])},
          useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i1.DOCUMENT}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.PLATFORM_ID}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i0.ɵe}}}],multi:false},module:{reference:i0.HttpClientXsrfModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.ɵe}},
          useClass:(null as any),useValue:'XSRF-TOKEN',useFactory:(null as any),useExisting:(undefined as any),
          deps:(undefined as any),multi:false},module:{reference:i0.HttpClientXsrfModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i0.ɵf}},
          useClass:(null as any),useValue:'X-XSRF-TOKEN',useFactory:(null as any),
          useExisting:(undefined as any),deps:(undefined as any),multi:false},module:{reference:i0.HttpClientXsrfModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],modules:[{reference:i0.HttpClientXsrfModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])}],exportedDirectives:([] as any[]),
      exportedPipes:([] as any[])},{summaryKind:3,type:{reference:i0.ɵh,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.HttpXsrfTokenExtractor}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.ɵf}}}],lifecycleHooks:([] as any[])}},{summaryKind:3,
      type:{reference:i0.ɵg,diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
          isOptional:false,token:{identifier:{reference:i1.DOCUMENT}}},{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.PLATFORM_ID}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i0.ɵe}}}],lifecycleHooks:([] as any[])}}];
}
export function HttpClientNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.HttpClient,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.HttpHandler}}}],
      lifecycleHooks:([] as any[])}}];
}
export function ɵaNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.ɵa,diDeps:([] as any[]),lifecycleHooks:([] as any[])}}];
}
export function JsonpClientBackendNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.JsonpClientBackend,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ɵb}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i1.DOCUMENT}}}],lifecycleHooks:([] as any[])}}];
}
export function JsonpInterceptorNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.JsonpInterceptor,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.JsonpClientBackend}}}],
      lifecycleHooks:([] as any[])}}];
}
export function ɵdNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.ɵd,diDeps:([] as any[]),lifecycleHooks:([] as any[])}}];
}
export function HttpXhrBackendNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.HttpXhrBackend,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.XhrFactory}}}],
      lifecycleHooks:([] as any[])}}];
}
export function ɵgNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.ɵg,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i1.DOCUMENT}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.PLATFORM_ID}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.ɵe}}}],
      lifecycleHooks:([] as any[])}}];
}
export function ɵhNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.ɵh,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.HttpXsrfTokenExtractor}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.ɵf}}}],lifecycleHooks:([] as any[])}}];
}
