// @angular/material/button/typings/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/material/button';
import * as i1 from '@angular/common';
import * as i2 from '@angular/core';
import * as i3 from '@angular/cdk/bidi';
import * as i4 from '@angular/platform-browser';
import * as i5 from '@angular/material/core';
import * as i6 from '@angular/cdk/platform';
import * as i7 from '@angular/cdk/a11y';
import * as i8 from '../../../common/common.ngsummary';
import * as i9 from '../../core/typings/index.ngsummary';
import * as i10 from '../../../cdk/bidi/typings/index.ngsummary';
import * as i11 from '../../../cdk/platform/typings/index.ngsummary';
import * as i12 from '../../../cdk/a11y/typings/index.ngsummary';
import * as i13 from './index.ngfactory';
export function MatButtonModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MatButtonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i1.NgLocalization}},
          useClass:{reference:i1.NgLocaleLocalization,diDeps:[{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
          multi:false},module:{reference:i1.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i3.DIR_DOCUMENT}},useClass:(null as any),
              useValue:(undefined as any),useFactory:(null as any),useExisting:{identifier:{reference:i4.DOCUMENT}},
              deps:(undefined as any),multi:false},module:{reference:i3.BidiModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i3.Directionality,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i3.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])}},useClass:{reference:i3.Directionality,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i3.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i3.DIR_DOCUMENT}}}],
              multi:false},module:{reference:i3.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i5.MATERIAL_SANITY_CHECKS}},useClass:(null as any),
              useValue:true,useFactory:(null as any),useExisting:(undefined as any),
              deps:(undefined as any),multi:false},module:{reference:i5.MatCommonModule,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i5.MATERIAL_SANITY_CHECKS}}}],
              lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i6.Platform,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},useClass:{reference:i6.Platform,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])},useValue:(undefined as any),
              useFactory:(null as any),useExisting:(undefined as any),deps:([] as any[]),
              multi:false},module:{reference:i6.PlatformModule,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i7.InteractivityChecker,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i6.Platform}}}],lifecycleHooks:([] as any[])}},
              useClass:{reference:i7.InteractivityChecker,diDeps:[{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}}],
                  lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}}],
              multi:false},module:{reference:i7.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i7.FocusTrapFactory,diDeps:[{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i7.InteractivityChecker}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i6.Platform}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}}],
              lifecycleHooks:([] as any[])}},useClass:{reference:i7.FocusTrapFactory,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i7.InteractivityChecker}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.Platform}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i2.NgZone}}}],lifecycleHooks:([] as any[])},
              useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
              deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i7.InteractivityChecker}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.Platform}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i2.NgZone}}}],multi:false},module:{reference:i7.A11yModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i7.AriaDescriber,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:false,token:{identifier:{reference:i6.Platform}}}],lifecycleHooks:[1]}},
              useClass:{reference:i7.AriaDescriber,diDeps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}}],
                  lifecycleHooks:[1]},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}}],
              multi:false},module:{reference:i7.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i7.LiveAnnouncer}},useClass:(null as any),
              useValue:(undefined as any),useFactory:{reference:i7.LIVE_ANNOUNCER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i7.LiveAnnouncer}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:true,token:{identifier:{reference:i7.LIVE_ANNOUNCER_ELEMENT_TOKEN}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i6.Platform}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i7.LiveAnnouncer}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
                      token:{identifier:{reference:i7.LIVE_ANNOUNCER_ELEMENT_TOKEN}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.Platform}}}],multi:false},module:{reference:i7.A11yModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i7.AriaDescriber}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i7.ARIA_DESCRIBER_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i7.AriaDescriber}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i6.Platform}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i7.AriaDescriber}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.Platform}}}],multi:false},module:{reference:i7.A11yModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i7.FocusMonitor}},
              useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i7.FOCUS_MONITOR_PROVIDER_FACTORY,
                  diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                      isOptional:true,token:{identifier:{reference:i7.FocusMonitor}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i2.NgZone}}},
                      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                          isOptional:false,token:{identifier:{reference:i6.Platform}}}]},
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i7.FocusMonitor}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i2.NgZone}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.Platform}}}],multi:false},module:{reference:i7.A11yModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],modules:[{reference:i1.CommonModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i5.CompatibilityModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i3.BidiModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i5.MatCommonModule,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
              token:{identifier:{reference:i5.MATERIAL_SANITY_CHECKS}}}],lifecycleHooks:([] as any[])},
          {reference:i6.PlatformModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i5.MatRippleModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i7.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i0.MatButtonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}],
      exportedDirectives:[{reference:i5.MatPrefixRejector},{reference:i5.MdPrefixRejector},
          {reference:i3.Dir},{reference:i0.MatButton},{reference:i0.MatAnchor},{reference:i0.MatMiniFab},
          {reference:i0.MatFab},{reference:i0.MatButtonCssMatStyler},{reference:i0.MatRaisedButtonCssMatStyler},
          {reference:i0.MatIconButtonCssMatStyler}],exportedPipes:([] as any[])},MatButtonNgSummary,
      MatAnchorNgSummary,MatMiniFabNgSummary,MatFabNgSummary,MatButtonCssMatStylerNgSummary,
      MatRaisedButtonCssMatStylerNgSummary,MatIconButtonCssMatStylerNgSummary,i8.CommonModuleNgSummary,
      i9.CompatibilityModuleNgSummary,i10.BidiModuleNgSummary,i9.MatCommonModuleNgSummary,
      i11.PlatformModuleNgSummary,i9.MatRippleModuleNgSummary,i12.A11yModuleNgSummary];
}
export function MatButtonNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatButton,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.Renderer2}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ElementRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i7.FocusMonitor}}}],lifecycleHooks:[1]},isComponent:true,
      selector:'button[mat-button], button[mat-raised-button], button[mat-icon-button],\n             button[mat-fab], button[mat-mini-fab]',
      exportAs:'matButton',inputs:{disabled:'disabled',disableRipple:'disableRipple',
          color:'color'},outputs:{},hostListeners:{},hostProperties:{disabled:'disabled || null'},
      hostAttributes:{},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:['*'],encapsulation:2},componentViewType:i13.View_MatButton_0,
      rendererType:i13.RenderType_MatButton,componentFactory:i13.MatButtonNgFactory}];
}
export function MatAnchorNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatAnchor,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i7.FocusMonitor}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.Renderer2}}}],lifecycleHooks:[1]},isComponent:true,
      selector:'a[mat-button], a[mat-raised-button], a[mat-icon-button], a[mat-fab], a[mat-mini-fab]',
      exportAs:'matButton, matAnchor',inputs:{disabled:'disabled',disableRipple:'disableRipple',
          color:'color'},outputs:{},hostListeners:{click:'_haltDisabledEvents($event)'},
      hostProperties:{'attr.tabindex':'disabled ? -1 : 0','attr.disabled':'disabled || null',
          'attr.aria-disabled':'disabled.toString()'},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:['*'],encapsulation:2},componentViewType:i13.View_MatAnchor_0,
      rendererType:i13.RenderType_MatAnchor,componentFactory:i13.MatAnchorNgFactory}];
}
export function MatMiniFabNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatMiniFab,diDeps:[{isAttribute:false,
      isHost:false,isSelf:true,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MatButton}}},
      {isAttribute:false,isHost:false,isSelf:true,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i0.MatAnchor}}}],lifecycleHooks:([] as any[])},
      isComponent:false,selector:'button[mat-mini-fab], a[mat-mini-fab]',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-mini-fab'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatFabNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatFab,diDeps:[{isAttribute:false,isHost:false,
      isSelf:true,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MatButton}}},
      {isAttribute:false,isHost:false,isSelf:true,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i0.MatAnchor}}}],lifecycleHooks:([] as any[])},
      isComponent:false,selector:'button[mat-fab], a[mat-fab]',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-fab'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatButtonCssMatStylerNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatButtonCssMatStyler,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:false,selector:'button[mat-button], a[mat-button]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-button'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatRaisedButtonCssMatStylerNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatRaisedButtonCssMatStyler,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:false,selector:'button[mat-raised-button], a[mat-raised-button]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-raised-button'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatIconButtonCssMatStylerNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatIconButtonCssMatStyler,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:false,selector:'button[mat-icon-button], a[mat-icon-button]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-icon-button'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
