// @angular/material/table/typings/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/material/table';
import * as i1 from '@angular/common';
import * as i2 from '@angular/core';
import * as i3 from '@angular/cdk/bidi';
import * as i4 from '@angular/platform-browser';
import * as i5 from '@angular/material/core';
import * as i6 from '@angular/cdk/table';
import * as i7 from '../../../common/common.ngsummary';
import * as i8 from '../../../cdk/table/typings/index.ngsummary';
import * as i9 from '../../core/typings/index.ngsummary';
import * as i10 from '../../../cdk/bidi/typings/index.ngsummary';
import * as i11 from './index.ngfactory';
export function MatTableModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MatTableModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i1.NgLocalization}},
          useClass:{reference:i1.NgLocaleLocalization,diDeps:[{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
          multi:false},module:{reference:i1.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i3.DIR_DOCUMENT}},useClass:(null as any),
              useValue:(undefined as any),useFactory:(null as any),useExisting:{identifier:{reference:i4.DOCUMENT}},
              deps:(undefined as any),multi:false},module:{reference:i3.BidiModule,
              diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i3.Directionality,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i3.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])}},useClass:{reference:i3.Directionality,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i3.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i3.DIR_DOCUMENT}}}],
              multi:false},module:{reference:i3.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i5.MATERIAL_SANITY_CHECKS}},useClass:(null as any),
              useValue:true,useFactory:(null as any),useExisting:(undefined as any),
              deps:(undefined as any),multi:false},module:{reference:i5.MatCommonModule,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i5.MATERIAL_SANITY_CHECKS}}}],
              lifecycleHooks:([] as any[])}}],modules:[{reference:i1.CommonModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i6.CdkTableModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i5.CompatibilityModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i3.BidiModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i5.MatCommonModule,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
              token:{identifier:{reference:i5.MATERIAL_SANITY_CHECKS}}}],lifecycleHooks:([] as any[])},
          {reference:i0.MatTableModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}],
      exportedDirectives:[{reference:i0.MatTable},{reference:i0.MatCellDef},{reference:i0.MatHeaderCellDef},
          {reference:i0.MatColumnDef},{reference:i0.MatHeaderCell},{reference:i0.MatCell},
          {reference:i0.MatHeaderRow},{reference:i0.MatRow},{reference:i0.MatHeaderRowDef},
          {reference:i0.MatRowDef}],exportedPipes:([] as any[])},MatTableNgSummary,
      MatCellDefNgSummary,MatHeaderCellDefNgSummary,MatColumnDefNgSummary,MatHeaderCellNgSummary,
      MatCellNgSummary,MatHeaderRowNgSummary,MatRowNgSummary,MatHeaderRowDefNgSummary,
      MatRowDefNgSummary,i7.CommonModuleNgSummary,i8.CdkTableModuleNgSummary,i9.CompatibilityModuleNgSummary,
      i10.BidiModuleNgSummary,i9.MatCommonModuleNgSummary];
}
export function MatTableNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatTable,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.IterableDiffers}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ChangeDetectorRef}}},{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.Renderer2}}},{isAttribute:true,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{value:'role'}}],lifecycleHooks:[0,
      1,4,5]},isComponent:true,selector:'mat-table',exportAs:'matTable',inputs:{trackBy:'trackBy',
      dataSource:'dataSource'},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-table'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:[{selectors:[{identifier:{reference:i6.CdkColumnDef}}],
          first:false,descendants:false,propertyName:'_columnDefs',read:(null as any)},
          {selectors:[{identifier:{reference:i6.CdkHeaderRowDef}}],first:true,descendants:true,
              propertyName:'_headerDef',read:(null as any)},{selectors:[{identifier:{reference:i6.CdkRowDef}}],
              first:false,descendants:false,propertyName:'_rowDefs',read:(null as any)}],
      viewQueries:[{selectors:[{identifier:{reference:i6.RowPlaceholder}}],first:true,
          descendants:true,propertyName:'_rowPlaceholder',read:(null as any)},{selectors:[{identifier:{reference:i6.HeaderRowPlaceholder}}],
          first:true,descendants:true,propertyName:'_headerRowPlaceholder',read:(null as any)}],
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:([] as any[]),encapsulation:2},componentViewType:i11.View_MatTable_0,
      rendererType:i11.RenderType_MatTable,componentFactory:i11.MatTableNgFactory}];
}
export function MatCellDefNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatCellDef,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.TemplateRef}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'[matCellDef]',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:[{token:{identifier:{reference:i6.CdkCellDef}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i0.MatCellDef}},deps:(undefined as any),
          multi:false}],viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatHeaderCellDefNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatHeaderCellDef,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.TemplateRef}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'[matHeaderCellDef]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{},providers:[{token:{identifier:{reference:i6.CdkHeaderCellDef}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i0.MatHeaderCellDef}},deps:(undefined as any),
          multi:false}],viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatColumnDefNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatColumnDef,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'[matColumnDef]',exportAs:(null as any),inputs:{name:'matColumnDef'},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:[{token:{identifier:{reference:i6.CdkColumnDef}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i0.MatColumnDef}},deps:(undefined as any),
          multi:false}],viewProviders:([] as any[]),queries:[{selectors:[{identifier:{reference:i6.CdkCellDef}}],
          first:true,descendants:true,propertyName:'cell',read:(null as any)},{selectors:[{identifier:{reference:i6.CdkHeaderCellDef}}],
          first:true,descendants:true,propertyName:'headerCell',read:(null as any)}],
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function MatHeaderCellNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatHeaderCell,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.CdkColumnDef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ElementRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.Renderer2}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'mat-header-cell',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-header-cell',
          role:'columnheader'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatCellNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatCell,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.CdkColumnDef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ElementRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.Renderer2}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'mat-cell',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-cell',
          role:'gridcell'},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
export function MatHeaderRowNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatHeaderRow,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:true,selector:'mat-header-row',exportAs:'matHeaderRow',inputs:{},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-header-row',
          role:'row'},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:([] as any[]),encapsulation:2},componentViewType:i11.View_MatHeaderRow_0,
      rendererType:i11.RenderType_MatHeaderRow,componentFactory:i11.MatHeaderRowNgFactory}];
}
export function MatRowNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatRow,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:true,selector:'mat-row',exportAs:'matRow',inputs:{},outputs:{},hostListeners:{},
      hostProperties:{},hostAttributes:{class:'mat-row',role:'row'},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:([] as any[]),encapsulation:2},componentViewType:i11.View_MatRow_0,
      rendererType:i11.RenderType_MatRow,componentFactory:i11.MatRowNgFactory}];
}
export function MatHeaderRowDefNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatHeaderRowDef,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.TemplateRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.IterableDiffers}}}],lifecycleHooks:[3]},
      isComponent:false,selector:'[matHeaderRowDef]',exportAs:(null as any),inputs:{columns:'matHeaderRowDef'},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:[{token:{identifier:{reference:i6.CdkHeaderRowDef}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i0.MatHeaderRowDef}},deps:(undefined as any),
          multi:false}],viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatRowDefNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatRowDef,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.TemplateRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.IterableDiffers}}}],lifecycleHooks:[3]},
      isComponent:false,selector:'[matRowDef]',exportAs:(null as any),inputs:{columns:'matRowDefColumns',
          when:'matRowDefWhen'},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},
      providers:[{token:{identifier:{reference:i6.CdkRowDef}},useClass:(null as any),
          useValue:(undefined as any),useFactory:(null as any),useExisting:{identifier:{reference:i0.MatRowDef}},
          deps:(undefined as any),multi:false}],viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:(null as any),
      template:(null as any),componentViewType:(null as any),rendererType:(null as any),
      componentFactory:(null as any)}];
}
