// @angular/material/card/typings/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/material/card';
import * as i1 from '@angular/cdk/bidi';
import * as i2 from '@angular/platform-browser';
import * as i3 from '@angular/material/core';
import * as i4 from '../../core/typings/index.ngsummary';
import * as i5 from '../../../cdk/bidi/typings/index.ngsummary';
import * as i6 from './index.ngfactory';
export function MatCardModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MatCardModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i1.DIR_DOCUMENT}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i2.DOCUMENT}},deps:(undefined as any),
          multi:false},module:{reference:i1.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i1.Directionality,diDeps:[{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])}},useClass:{reference:i1.Directionality,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
              useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
              multi:false},module:{reference:i1.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          {provider:{token:{identifier:{reference:i3.MATERIAL_SANITY_CHECKS}},useClass:(null as any),
              useValue:true,useFactory:(null as any),useExisting:(undefined as any),
              deps:(undefined as any),multi:false},module:{reference:i3.MatCommonModule,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,
                  isOptional:true,token:{identifier:{reference:i3.MATERIAL_SANITY_CHECKS}}}],
              lifecycleHooks:([] as any[])}}],modules:[{reference:i3.CompatibilityModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i1.BidiModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i3.MatCommonModule,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
              token:{identifier:{reference:i3.MATERIAL_SANITY_CHECKS}}}],lifecycleHooks:([] as any[])},
          {reference:i0.MatCardModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}],
      exportedDirectives:[{reference:i3.MatPrefixRejector},{reference:i3.MdPrefixRejector},
          {reference:i1.Dir},{reference:i0.MatCard},{reference:i0.MatCardHeader},{reference:i0.MatCardTitleGroup},
          {reference:i0.MatCardContent},{reference:i0.MatCardTitle},{reference:i0.MatCardSubtitle},
          {reference:i0.MatCardActions},{reference:i0.MatCardFooter},{reference:i0.MatCardSmImage},
          {reference:i0.MatCardMdImage},{reference:i0.MatCardLgImage},{reference:i0.MatCardImage},
          {reference:i0.MatCardXlImage},{reference:i0.MatCardAvatar}],exportedPipes:([] as any[])},
      MatCardNgSummary,MatCardHeaderNgSummary,MatCardTitleGroupNgSummary,MatCardContentNgSummary,
      MatCardTitleNgSummary,MatCardSubtitleNgSummary,MatCardActionsNgSummary,MatCardFooterNgSummary,
      MatCardSmImageNgSummary,MatCardMdImageNgSummary,MatCardLgImageNgSummary,MatCardImageNgSummary,
      MatCardXlImageNgSummary,MatCardAvatarNgSummary,i4.CompatibilityModuleNgSummary,
      i5.BidiModuleNgSummary,i4.MatCommonModuleNgSummary];
}
export function MatCardNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatCard,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:true,selector:'mat-card',exportAs:'matCard',inputs:{},outputs:{},
      hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-card'},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:['*','mat-card-footer'],encapsulation:2},componentViewType:i6.View_MatCard_0,
      rendererType:i6.RenderType_MatCard,componentFactory:i6.MatCardNgFactory}];
}
export function MatCardHeaderNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatCardHeader,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:true,selector:'mat-card-header',exportAs:(null as any),inputs:{},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-card-header'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:['[mat-card-avatar], [matCardAvatar]','mat-card-title, mat-card-subtitle, [mat-card-title], [mat-card-subtitle]',
              '*'],encapsulation:2},componentViewType:i6.View_MatCardHeader_0,rendererType:i6.RenderType_MatCardHeader,
      componentFactory:i6.MatCardHeaderNgFactory}];
}
export function MatCardTitleGroupNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatCardTitleGroup,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:true,selector:'mat-card-title-group',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-card-title-group'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:0,template:{animations:([] as any[]),ngContentSelectors:['mat-card-title, mat-card-subtitle, [mat-card-title], [mat-card-subtitle]',
          'img','*'],encapsulation:2},componentViewType:i6.View_MatCardTitleGroup_0,
      rendererType:i6.RenderType_MatCardTitleGroup,componentFactory:i6.MatCardTitleGroupNgFactory}];
}
export function MatCardContentNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatCardContent,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'mat-card-content',exportAs:(null as any),inputs:{},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-card-content'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatCardTitleNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatCardTitle,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'mat-card-title, [mat-card-title], [matCardTitle]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-card-title'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatCardSubtitleNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatCardSubtitle,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'mat-card-subtitle, [mat-card-subtitle], [matCardSubtitle]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-card-subtitle'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatCardActionsNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatCardActions,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'mat-card-actions',exportAs:'matCardActions',inputs:{align:'align'},
      outputs:{},hostListeners:{},hostProperties:{'class.mat-card-actions-align-end':'align === "end"'},
      hostAttributes:{class:'mat-card-actions'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:([] as any[]),viewQueries:([] as any[]),entryComponents:([] as any[]),
      changeDetection:(null as any),template:(null as any),componentViewType:(null as any),
      rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatCardFooterNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatCardFooter,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'mat-card-footer',exportAs:(null as any),inputs:{},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-card-footer'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatCardSmImageNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatCardSmImage,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'[mat-card-sm-image], [matCardImageSmall]',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-card-sm-image'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatCardMdImageNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatCardMdImage,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'[mat-card-md-image], [matCardImageMedium]',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-card-md-image'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatCardLgImageNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatCardLgImage,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'[mat-card-lg-image], [matCardImageLarge]',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-card-lg-image'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatCardImageNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatCardImage,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'[mat-card-image], [matCardImage]',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-card-image'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatCardXlImageNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatCardXlImage,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'[mat-card-xl-image], [matCardImageXLarge]',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-card-xl-image'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatCardAvatarNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatCardAvatar,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'[mat-card-avatar], [matCardAvatar]',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-card-avatar'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
