// @angular/material/sidenav/typings/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/material/sidenav';
import * as i1 from '@angular/common';
import * as i2 from '@angular/core';
import * as i3 from '@angular/cdk/bidi';
import * as i4 from '@angular/platform-browser';
import * as i5 from '@angular/material/core';
import * as i6 from '@angular/cdk/platform';
import * as i7 from '@angular/cdk/a11y';
import * as i8 from '@angular/cdk/scrolling';
import * as i9 from '@angular/cdk/overlay';
import * as i10 from '@angular/cdk/portal';
import * as i11 from '../../../common/common.ngsummary';
import * as i12 from '../../core/typings/index.ngsummary';
import * as i13 from '../../../cdk/bidi/typings/index.ngsummary';
import * as i14 from '../../../cdk/platform/typings/index.ngsummary';
import * as i15 from '../../../cdk/a11y/typings/index.ngsummary';
import * as i16 from '../../../cdk/portal/typings/index.ngsummary';
import * as i17 from '../../../cdk/scrolling/typings/index.ngsummary';
import * as i18 from '../../../cdk/overlay/typings/index.ngsummary';
import * as i19 from './index.ngfactory';
export function MatSidenavModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MatSidenavModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i1.NgLocalization}},
      useClass:{reference:i1.NgLocaleLocalization,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
      useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
          isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
      multi:false},module:{reference:i1.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i3.DIR_DOCUMENT}},useClass:(null as any),
          useValue:(undefined as any),useFactory:(null as any),useExisting:{identifier:{reference:i4.DOCUMENT}},
          deps:(undefined as any),multi:false},module:{reference:i3.BidiModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i3.Directionality,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
              token:{identifier:{reference:i3.DIR_DOCUMENT}}}],lifecycleHooks:([] as any[])}},
          useClass:{reference:i3.Directionality,diDeps:[{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i3.DIR_DOCUMENT}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:true,token:{identifier:{reference:i3.DIR_DOCUMENT}}}],
          multi:false},module:{reference:i3.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i5.MATERIAL_SANITY_CHECKS}},useClass:(null as any),
          useValue:true,useFactory:(null as any),useExisting:(undefined as any),deps:(undefined as any),
          multi:false},module:{reference:i5.MatCommonModule,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i5.MATERIAL_SANITY_CHECKS}}}],
          lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i6.Platform,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},useClass:{reference:i6.Platform,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},useValue:(undefined as any),
          useFactory:(null as any),useExisting:(undefined as any),deps:([] as any[]),
          multi:false},module:{reference:i6.PlatformModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i7.InteractivityChecker,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}}],
          lifecycleHooks:([] as any[])}},useClass:{reference:i7.InteractivityChecker,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i6.Platform}}}],lifecycleHooks:([] as any[])},
          useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i6.Platform}}}],multi:false},module:{reference:i7.A11yModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i7.FocusTrapFactory,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i7.InteractivityChecker}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i2.NgZone}}}],lifecycleHooks:([] as any[])}},
          useClass:{reference:i7.FocusTrapFactory,diDeps:[{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i7.InteractivityChecker}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i6.Platform}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i7.InteractivityChecker}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i6.Platform}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}}],
          multi:false},module:{reference:i7.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i7.AriaDescriber,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}}],
          lifecycleHooks:[1]}},useClass:{reference:i7.AriaDescriber,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}}],
          lifecycleHooks:[1]},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}}],
          multi:false},module:{reference:i7.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i7.LiveAnnouncer}},useClass:(null as any),
          useValue:(undefined as any),useFactory:{reference:i7.LIVE_ANNOUNCER_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i7.LiveAnnouncer}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
                      token:{identifier:{reference:i7.LIVE_ANNOUNCER_ELEMENT_TOKEN}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.Platform}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i7.LiveAnnouncer}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i7.LIVE_ANNOUNCER_ELEMENT_TOKEN}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i6.Platform}}}],multi:false},module:{reference:i7.A11yModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i7.AriaDescriber}},
          useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i7.ARIA_DESCRIBER_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i7.AriaDescriber}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.Platform}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i7.AriaDescriber}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}}],
          multi:false},module:{reference:i7.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i7.FocusMonitor}},useClass:(null as any),
          useValue:(undefined as any),useFactory:{reference:i7.FOCUS_MONITOR_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i7.FocusMonitor}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i2.NgZone}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.Platform}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i7.FocusMonitor}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i6.Platform}}}],multi:false},module:{reference:i7.A11yModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i8.ScrollDispatcher}},
          useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i8.SCROLL_DISPATCHER_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i8.ScrollDispatcher}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i2.NgZone}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.Platform}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i8.ScrollDispatcher}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i6.Platform}}}],multi:false},module:{reference:i8.ScrollDispatchModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i9.Overlay,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i9.ScrollStrategyOptions}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i9.OverlayContainer}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i2.ComponentFactoryResolver}}},{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i9.ɵf}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i2.ApplicationRef}}},{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.Injector}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i2.NgZone}}}],lifecycleHooks:([] as any[])}},
          useClass:{reference:i9.Overlay,diDeps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i9.ScrollStrategyOptions}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i9.OverlayContainer}}},{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ComponentFactoryResolver}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i9.ɵf}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ApplicationRef}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i2.Injector}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i9.ScrollStrategyOptions}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i9.OverlayContainer}}},{isAttribute:false,
                  isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ComponentFactoryResolver}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i9.ɵf}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ApplicationRef}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i2.Injector}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}}],
          multi:false},module:{reference:i9.OverlayModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i9.ɵf,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i8.ViewportRuler}}}],
          lifecycleHooks:([] as any[])}},useClass:{reference:i9.ɵf,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i8.ViewportRuler}}}],
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i8.ViewportRuler}}}],
          multi:false},module:{reference:i9.OverlayModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i8.ViewportRuler}},useClass:(null as any),
          useValue:(undefined as any),useFactory:{reference:i8.VIEWPORT_RULER_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i8.ViewportRuler}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.Platform}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i2.NgZone}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i8.ScrollDispatcher}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i8.ViewportRuler}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i2.NgZone}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i8.ScrollDispatcher}}}],
          multi:false},module:{reference:i9.OverlayModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i9.OverlayContainer}},useClass:(null as any),
          useValue:(undefined as any),useFactory:{reference:i9.ɵa,diDeps:[{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,token:{identifier:{reference:i9.OverlayContainer}}}]},
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:true,isOptional:true,token:{identifier:{reference:i9.OverlayContainer}}}],
          multi:false},module:{reference:i9.OverlayModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i9.ɵc}},useClass:(null as any),useValue:(undefined as any),
          useFactory:{reference:i9.ɵd,diDeps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i9.Overlay}}}]},
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i9.Overlay}}}],
          multi:false},module:{reference:i9.OverlayModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i9.ScrollStrategyOptions,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i8.ScrollDispatcher}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i8.ViewportRuler}}}],lifecycleHooks:([] as any[])}},
          useClass:{reference:i9.ScrollStrategyOptions,diDeps:[{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i8.ScrollDispatcher}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i8.ViewportRuler}}}],lifecycleHooks:([] as any[])},
          useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i8.ScrollDispatcher}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i8.ViewportRuler}}}],
          multi:false},module:{reference:i9.OverlayModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}}],
      modules:[{reference:i1.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i5.CompatibilityModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i3.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          {reference:i5.MatCommonModule,diDeps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:true,token:{identifier:{reference:i5.MATERIAL_SANITY_CHECKS}}}],
              lifecycleHooks:([] as any[])},{reference:i6.PlatformModule,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])},{reference:i7.A11yModule,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])},{reference:i10.PortalModule,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])},{reference:i8.ScrollDispatchModule,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])},{reference:i9.OverlayModule,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])},{reference:i0.MatSidenavModule,diDeps:([] as any[]),
              lifecycleHooks:([] as any[])}],exportedDirectives:[{reference:i5.MatPrefixRejector},
          {reference:i5.MdPrefixRejector},{reference:i3.Dir},{reference:i0.MatDrawer},
          {reference:i0.MatDrawerContainer},{reference:i0.MatDrawerContent},{reference:i0.MatSidenav},
          {reference:i0.MatSidenavContainer},{reference:i0.MatSidenavContent}],exportedPipes:([] as any[])},
      MatDrawerNgSummary,MatDrawerContainerNgSummary,MatDrawerContentNgSummary,MatSidenavNgSummary,
      MatSidenavContainerNgSummary,MatSidenavContentNgSummary,i11.CommonModuleNgSummary,
      i12.CompatibilityModuleNgSummary,i13.BidiModuleNgSummary,i12.MatCommonModuleNgSummary,
      i14.PlatformModuleNgSummary,i15.A11yModuleNgSummary,i16.PortalModuleNgSummary,
      i17.ScrollDispatchModuleNgSummary,i18.OverlayModuleNgSummary];
}
export function MatDrawerNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatDrawer,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i7.FocusTrapFactory}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i4.DOCUMENT}}}],
      lifecycleHooks:[1,4]},isComponent:true,selector:'mat-drawer',exportAs:'matDrawer',
      inputs:{position:'position',align:'align',mode:'mode',disableClose:'disableClose',
          opened:'opened'},outputs:{onOpen:'open',onClose:'close',onPositionChanged:'positionChanged',
          onAlignChanged:'align-changed'},hostListeners:{'@transform.start':'_onAnimationStart($event)',
          '@transform.done':'_onAnimationEnd($event)',keydown:'handleKeydown($event)'},
      hostProperties:{'@transform':'_animationState','attr.align':'null','class.mat-drawer-end':'position === "end"',
          'class.mat-drawer-over':'mode === "over"','class.mat-drawer-push':'mode === "push"',
          'class.mat-drawer-side':'mode === "side"'},hostAttributes:{class:'mat-drawer',
          tabIndex:'-1'},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:0,template:{animations:['transform'],
          ngContentSelectors:['*'],encapsulation:2},componentViewType:i19.View_MatDrawer_0,
      rendererType:i19.RenderType_MatDrawer,componentFactory:i19.MatDrawerNgFactory}];
}
export function MatDrawerContainerNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatDrawerContainer,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i3.Directionality}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ElementRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.Renderer2}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.NgZone}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ChangeDetectorRef}}}],
      lifecycleHooks:[1,4]},isComponent:true,selector:'mat-drawer-container',exportAs:'matDrawerContainer',
      inputs:{},outputs:{backdropClick:'backdropClick'},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-drawer-container'},providers:([] as any[]),viewProviders:([] as any[]),
      queries:[{selectors:[{identifier:{reference:i0.MatDrawer}}],first:false,descendants:false,
          propertyName:'_drawers',read:(null as any)},{selectors:[{identifier:{reference:i0.MatDrawerContent}}],
          first:true,descendants:true,propertyName:'_content',read:(null as any)}],
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:['mat-drawer','mat-drawer-content','*'],encapsulation:2},
      componentViewType:i19.View_MatDrawerContainer_0,rendererType:i19.RenderType_MatDrawerContainer,
      componentFactory:i19.MatDrawerContainerNgFactory}];
}
export function MatDrawerContentNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatDrawerContent,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ChangeDetectorRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.MatDrawerContainer}}}],lifecycleHooks:[4]},
      isComponent:true,selector:'mat-drawer-content',exportAs:(null as any),inputs:{},
      outputs:{},hostListeners:{},hostProperties:{'style.marginLeft.px':'_margins.left',
          'style.marginRight.px':'_margins.right'},hostAttributes:{class:'mat-drawer-content'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:['*'],encapsulation:2},componentViewType:i19.View_MatDrawerContent_0,
      rendererType:i19.RenderType_MatDrawerContent,componentFactory:i19.MatDrawerContentNgFactory}];
}
export function MatSidenavNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatSidenav,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i7.FocusTrapFactory}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i4.DOCUMENT}}}],
      lifecycleHooks:[1,4]},isComponent:true,selector:'mat-sidenav',exportAs:'matSidenav',
      inputs:{position:'position',align:'align',mode:'mode',disableClose:'disableClose',
          opened:'opened',fixedInViewport:'fixedInViewport',fixedTopGap:'fixedTopGap',
          fixedBottomGap:'fixedBottomGap'},outputs:{onOpen:'open',onClose:'close',
          onPositionChanged:'positionChanged',onAlignChanged:'align-changed'},hostListeners:{'@transform.start':'_onAnimationStart($event)',
          '@transform.done':'_onAnimationEnd($event)',keydown:'handleKeydown($event)'},
      hostProperties:{'@transform':'_animationState','attr.align':'null','class.mat-drawer-end':'position === "end"',
          'class.mat-drawer-over':'mode === "over"','class.mat-drawer-push':'mode === "push"',
          'class.mat-drawer-side':'mode === "side"','class.mat-sidenav-fixed':'fixedInViewport',
          'style.top.px':'fixedInViewport ? fixedTopGap : null','style.bottom.px':'fixedInViewport ? fixedBottomGap : null'},
      hostAttributes:{class:'mat-drawer mat-sidenav',tabIndex:'-1'},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:['transform'],
          ngContentSelectors:['*'],encapsulation:2},componentViewType:i19.View_MatSidenav_0,
      rendererType:i19.RenderType_MatSidenav,componentFactory:i19.MatSidenavNgFactory}];
}
export function MatSidenavContainerNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatSidenavContainer,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i3.Directionality}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ElementRef}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.Renderer2}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.NgZone}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ChangeDetectorRef}}}],
      lifecycleHooks:[1,4]},isComponent:true,selector:'mat-sidenav-container',exportAs:'matSidenavContainer',
      inputs:{},outputs:{backdropClick:'backdropClick'},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-drawer-container mat-sidenav-container'},providers:([] as any[]),
      viewProviders:([] as any[]),queries:[{selectors:[{identifier:{reference:i0.MatSidenav}}],
          first:false,descendants:false,propertyName:'_drawers',read:(null as any)},
          {selectors:[{identifier:{reference:i0.MatSidenavContent}}],first:true,descendants:true,
              propertyName:'_content',read:(null as any)}],viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:['mat-sidenav','mat-sidenav-content','*'],encapsulation:2},
      componentViewType:i19.View_MatSidenavContainer_0,rendererType:i19.RenderType_MatSidenavContainer,
      componentFactory:i19.MatSidenavContainerNgFactory}];
}
export function MatSidenavContentNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatSidenavContent,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ChangeDetectorRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.MatSidenavContainer}}}],lifecycleHooks:[4]},
      isComponent:true,selector:'mat-sidenav-content',exportAs:(null as any),inputs:{},
      outputs:{},hostListeners:{},hostProperties:{'style.marginLeft.px':'_margins.left',
          'style.marginRight.px':'_margins.right'},hostAttributes:{class:'mat-drawer-content mat-sidenav-content'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:['*'],encapsulation:2},componentViewType:i19.View_MatSidenavContent_0,
      rendererType:i19.RenderType_MatSidenavContent,componentFactory:i19.MatSidenavContentNgFactory}];
}
