// @angular/material/stepper/typings/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/material/stepper';
import * as i1 from '@angular/cdk/bidi';
import * as i2 from '@angular/platform-browser';
import * as i3 from '@angular/material/core';
import * as i4 from '@angular/common';
import * as i5 from '@angular/core';
import * as i6 from '@angular/cdk/platform';
import * as i7 from '@angular/cdk/a11y';
import * as i8 from '@angular/material/icon';
import * as i9 from '@angular/http';
import * as i10 from '@angular/cdk/portal';
import * as i11 from '@angular/material/button';
import * as i12 from '@angular/cdk/stepper';
import * as i13 from '../../core/typings/index.ngsummary';
import * as i14 from '../../../cdk/bidi/typings/index.ngsummary';
import * as i15 from '../../../common/common.ngsummary';
import * as i16 from '../../../cdk/portal/typings/index.ngsummary';
import * as i17 from '../../../cdk/platform/typings/index.ngsummary';
import * as i18 from '../../../cdk/a11y/typings/index.ngsummary';
import * as i19 from '../../button/typings/index.ngsummary';
import * as i20 from '../../../cdk/stepper/typings/index.ngsummary';
import * as i21 from '../../icon/typings/index.ngsummary';
import * as i22 from './index.ngfactory';
export function MatStepperModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MatStepperModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i1.DIR_DOCUMENT}},
      useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
      useExisting:{identifier:{reference:i2.DOCUMENT}},deps:(undefined as any),multi:false},
      module:{reference:i1.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i1.Directionality,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
          lifecycleHooks:([] as any[])}},useClass:{reference:i1.Directionality,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.DIR_DOCUMENT}}}],
          multi:false},module:{reference:i1.BidiModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i3.MATERIAL_SANITY_CHECKS}},useClass:(null as any),
          useValue:true,useFactory:(null as any),useExisting:(undefined as any),deps:(undefined as any),
          multi:false},module:{reference:i3.MatCommonModule,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i3.MATERIAL_SANITY_CHECKS}}}],
          lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i4.NgLocalization}},
          useClass:{reference:i4.NgLocaleLocalization,diDeps:[{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.LOCALE_ID}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.LOCALE_ID}}}],
          multi:false},module:{reference:i4.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i6.Platform,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          useClass:{reference:i6.Platform,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
          deps:([] as any[]),multi:false},module:{reference:i6.PlatformModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i7.InteractivityChecker,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i6.Platform}}}],lifecycleHooks:([] as any[])}},
          useClass:{reference:i7.InteractivityChecker,diDeps:[{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}}],
          multi:false},module:{reference:i7.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i7.FocusTrapFactory,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i7.InteractivityChecker}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i6.Platform}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.NgZone}}}],
          lifecycleHooks:([] as any[])}},useClass:{reference:i7.FocusTrapFactory,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i7.InteractivityChecker}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i6.Platform}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.NgZone}}}],
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i7.InteractivityChecker}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i6.Platform}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.NgZone}}}],
          multi:false},module:{reference:i7.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i7.AriaDescriber,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}}],
          lifecycleHooks:[1]}},useClass:{reference:i7.AriaDescriber,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}}],
          lifecycleHooks:[1]},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}}],
          multi:false},module:{reference:i7.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i7.LiveAnnouncer}},useClass:(null as any),
          useValue:(undefined as any),useFactory:{reference:i7.LIVE_ANNOUNCER_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i7.LiveAnnouncer}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
                      token:{identifier:{reference:i7.LIVE_ANNOUNCER_ELEMENT_TOKEN}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.Platform}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i7.LiveAnnouncer}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i7.LIVE_ANNOUNCER_ELEMENT_TOKEN}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i6.Platform}}}],multi:false},module:{reference:i7.A11yModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i7.AriaDescriber}},
          useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i7.ARIA_DESCRIBER_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i7.AriaDescriber}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.Platform}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i7.AriaDescriber}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i6.Platform}}}],
          multi:false},module:{reference:i7.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i7.FocusMonitor}},useClass:(null as any),
          useValue:(undefined as any),useFactory:{reference:i7.FOCUS_MONITOR_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i7.FocusMonitor}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i5.NgZone}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i6.Platform}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i7.FocusMonitor}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.NgZone}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i6.Platform}}}],multi:false},module:{reference:i7.A11yModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i8.MatIconRegistry}},
          useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i8.ICON_REGISTRY_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i8.MatIconRegistry}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
                      token:{identifier:{reference:i9.Http}}},{isAttribute:false,isHost:false,
                      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.DomSanitizer}}}]},
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:true,isOptional:true,token:{identifier:{reference:i8.MatIconRegistry}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
                  token:{identifier:{reference:i9.Http}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.DomSanitizer}}}],
          multi:false},module:{reference:i8.MatIconModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i0.MatStepperIntl,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}},useClass:{reference:i0.MatStepperIntl,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:([] as any[]),multi:false},module:{reference:i0.MatStepperModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i3.ErrorStateMatcher,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},useClass:{reference:i3.ErrorStateMatcher,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])},useValue:(undefined as any),
          useFactory:(null as any),useExisting:(undefined as any),deps:([] as any[]),
          multi:false},module:{reference:i0.MatStepperModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}}],modules:[{reference:i3.CompatibilityModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i1.BidiModule,
      diDeps:([] as any[]),lifecycleHooks:([] as any[])},{reference:i3.MatCommonModule,
      diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
          token:{identifier:{reference:i3.MATERIAL_SANITY_CHECKS}}}],lifecycleHooks:([] as any[])},
      {reference:i4.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      {reference:i10.PortalModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      {reference:i6.PlatformModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      {reference:i3.MatRippleModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      {reference:i7.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      {reference:i11.MatButtonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      {reference:i12.CdkStepperModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      {reference:i8.MatIconModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      {reference:i0.MatStepperModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}],
      exportedDirectives:[{reference:i3.MatPrefixRejector},{reference:i3.MdPrefixRejector},
          {reference:i1.Dir},{reference:i0.MatHorizontalStepper},{reference:i0.MatVerticalStepper},
          {reference:i0.MatStep},{reference:i0.MatStepLabel},{reference:i0.MatStepper},
          {reference:i0.MatStepperNext},{reference:i0.MatStepperPrevious},{reference:i0.MatStepHeader}],
      exportedPipes:([] as any[])},MatHorizontalStepperNgSummary,MatVerticalStepperNgSummary,
      MatStepNgSummary,MatStepLabelNgSummary,MatStepperNgSummary,MatStepperNextNgSummary,
      MatStepperPreviousNgSummary,MatStepHeaderNgSummary,i13.CompatibilityModuleNgSummary,
      i14.BidiModuleNgSummary,i13.MatCommonModuleNgSummary,i15.CommonModuleNgSummary,
      i16.PortalModuleNgSummary,i17.PlatformModuleNgSummary,i13.MatRippleModuleNgSummary,
      i18.A11yModuleNgSummary,i19.MatButtonModuleNgSummary,i20.CdkStepperModuleNgSummary,
      i21.MatIconModuleNgSummary,{summaryKind:3,type:{reference:i0.MatStepperIntl,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{summaryKind:3,type:{reference:i3.ErrorStateMatcher,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}}];
}
export function MatHorizontalStepperNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatHorizontalStepper,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.Directionality}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i5.ChangeDetectorRef}}}],lifecycleHooks:([] as any[])},
      isComponent:true,selector:'mat-horizontal-stepper',exportAs:'matHorizontalStepper',
      inputs:{linear:'linear',selectedIndex:'selectedIndex',selected:'selected'},outputs:{selectionChange:'selectionChange'},
      hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-stepper-horizontal',
          role:'tablist'},providers:[{token:{identifier:{reference:i0.MatStepper}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i0.MatHorizontalStepper}},deps:(undefined as any),
          multi:false}],viewProviders:([] as any[]),queries:[{selectors:[{identifier:{reference:i0.MatStep}}],
          first:false,descendants:false,propertyName:'_steps',read:(null as any)}],
      viewQueries:[{selectors:[{identifier:{reference:i0.MatStepHeader}}],first:false,
          descendants:true,propertyName:'_stepHeader',read:{identifier:{reference:i5.ElementRef}}}],
      entryComponents:([] as any[]),changeDetection:0,template:{animations:['stepTransition'],
          ngContentSelectors:([] as any[]),encapsulation:2},componentViewType:i22.View_MatHorizontalStepper_0,
      rendererType:i22.RenderType_MatHorizontalStepper,componentFactory:i22.MatHorizontalStepperNgFactory}];
}
export function MatVerticalStepperNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatVerticalStepper,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.Directionality}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i5.ChangeDetectorRef}}}],lifecycleHooks:([] as any[])},
      isComponent:true,selector:'mat-vertical-stepper',exportAs:'matVerticalStepper',
      inputs:{linear:'linear',selectedIndex:'selectedIndex',selected:'selected'},outputs:{selectionChange:'selectionChange'},
      hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-stepper-vertical',
          role:'tablist'},providers:[{token:{identifier:{reference:i0.MatStepper}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i0.MatVerticalStepper}},deps:(undefined as any),
          multi:false}],viewProviders:([] as any[]),queries:[{selectors:[{identifier:{reference:i0.MatStep}}],
          first:false,descendants:false,propertyName:'_steps',read:(null as any)}],
      viewQueries:[{selectors:[{identifier:{reference:i0.MatStepHeader}}],first:false,
          descendants:true,propertyName:'_stepHeader',read:{identifier:{reference:i5.ElementRef}}}],
      entryComponents:([] as any[]),changeDetection:0,template:{animations:['stepTransition'],
          ngContentSelectors:([] as any[]),encapsulation:2},componentViewType:i22.View_MatVerticalStepper_0,
      rendererType:i22.RenderType_MatVerticalStepper,componentFactory:i22.MatVerticalStepperNgFactory}];
}
export function MatStepNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatStep,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.MatStepper}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:false,
          token:{identifier:{reference:i3.ErrorStateMatcher}}}],lifecycleHooks:[3]},
      isComponent:true,selector:'mat-step',exportAs:'matStep',inputs:{stepControl:'stepControl',
          label:'label',editable:'editable',optional:'optional',completed:'completed'},
      outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:[{token:{identifier:{reference:i3.ErrorStateMatcher}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i0.MatStep}},deps:(undefined as any),
          multi:false}],viewProviders:([] as any[]),queries:[{selectors:[{identifier:{reference:i0.MatStepLabel}}],
          first:true,descendants:true,propertyName:'stepLabel',read:(null as any)}],
      viewQueries:[{selectors:[{identifier:{reference:i5.TemplateRef}}],first:true,
          descendants:true,propertyName:'content',read:(null as any)}],entryComponents:([] as any[]),
      changeDetection:0,template:{animations:([] as any[]),ngContentSelectors:['*'],
          encapsulation:2},componentViewType:i22.View_MatStep_0,rendererType:i22.RenderType_MatStep,
      componentFactory:i22.MatStepNgFactory}];
}
export function MatStepLabelNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatStepLabel,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.TemplateRef}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'[matStepLabel]',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatStepperNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatStepper,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i1.Directionality}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i5.ChangeDetectorRef}}}],lifecycleHooks:([] as any[])},
      isComponent:false,selector:'[matStepper]',exportAs:(null as any),inputs:{linear:'linear',
          selectedIndex:'selectedIndex',selected:'selected'},outputs:{selectionChange:'selectionChange'},
      hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:[{selectors:[{identifier:{reference:i0.MatStep}}],
          first:false,descendants:false,propertyName:'_steps',read:(null as any)}],
      viewQueries:[{selectors:[{identifier:{reference:i0.MatStepHeader}}],first:false,
          descendants:true,propertyName:'_stepHeader',read:{identifier:{reference:i5.ElementRef}}}],
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatStepperNextNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatStepperNext,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i12.CdkStepper}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'button[matStepperNext]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{click:'_stepper.next()'},
      hostProperties:{},hostAttributes:{},providers:[{token:{identifier:{reference:i12.CdkStepper}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i0.MatStepper}},deps:(undefined as any),
          multi:false}],viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatStepperPreviousNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatStepperPrevious,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i12.CdkStepper}}}],
      lifecycleHooks:([] as any[])},isComponent:false,selector:'button[matStepperPrevious]',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{click:'_stepper.previous()'},
      hostProperties:{},hostAttributes:{},providers:[{token:{identifier:{reference:i12.CdkStepper}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i0.MatStepper}},deps:(undefined as any),
          multi:false}],viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatStepHeaderNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatStepHeader,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.MatStepperIntl}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i7.FocusMonitor}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i5.Renderer2}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.ChangeDetectorRef}}}],
      lifecycleHooks:[1]},isComponent:true,selector:'mat-step-header',exportAs:(null as any),
      inputs:{icon:'icon',label:'label',index:'index',selected:'selected',active:'active',
          optional:'optional'},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-step-header',
          role:'tab'},providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),
      viewQueries:([] as any[]),entryComponents:([] as any[]),changeDetection:0,template:{animations:([] as any[]),
          ngContentSelectors:([] as any[]),encapsulation:2},componentViewType:i22.View_MatStepHeader_0,
      rendererType:i22.RenderType_MatStepHeader,componentFactory:i22.MatStepHeaderNgFactory}];
}
export function MatStepperIntlNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.MatStepperIntl,diDeps:([] as any[]),lifecycleHooks:([] as any[])}}];
}
