// @angular/cdk/stepper/typings/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/cdk/stepper';
import * as i2 from '@angular/cdk/bidi';
import * as i3 from '@angular/platform-browser';
import * as i4 from '@angular/common';
export const CdkStepperModuleNgFactory:i0.NgModuleFactory<i1.CdkStepperModule> = i0.ɵcmf(i1.CdkStepperModule,
    ([] as any[]),(_l:any) => {
      return i0.ɵmod([i0.ɵmpd(512,i0.ComponentFactoryResolver,i0.ɵCodegenComponentFactoryResolver,
          [[8,([] as any[])],[3,i0.ComponentFactoryResolver],i0.NgModuleRef]),i0.ɵmpd(6144,
          i2.DIR_DOCUMENT,(null as any),[i3.DOCUMENT]),i0.ɵmpd(4608,i2.Directionality,
          i2.Directionality,[[2,i2.DIR_DOCUMENT]]),i0.ɵmpd(4608,i4.NgLocalization,
          i4.NgLocaleLocalization,[i0.LOCALE_ID]),i0.ɵmpd(512,i2.BidiModule,i2.BidiModule,
          ([] as any[])),i0.ɵmpd(512,i4.CommonModule,i4.CommonModule,([] as any[])),
          i0.ɵmpd(512,i1.CdkStepperModule,i1.CdkStepperModule,([] as any[]))]);
    });
const styles_CdkStep:any[] = ([] as any[]);
export const RenderType_CdkStep:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_CdkStep,
    data:{}});
function View_CdkStep_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵncd((null as any),0),(_l()(),i0.ɵand(0,(null as any),(null as any),
      0))],(null as any),(null as any));
}
export function View_CdkStep_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(2,[i0.ɵqud(402653184,1,{content:0}),(_l()(),i0.ɵand(0,[[1,2]],(null as any),
      0,(null as any),View_CdkStep_1))],(null as any),(null as any));
}
export function View_CdkStep_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'cdk-step',([] as any[]),
      (null as any),(null as any),(null as any),View_CdkStep_0,RenderType_CdkStep)),
      i0.ɵdid(573440,(null as any),1,i1.CdkStep,[i1.CdkStepper],(null as any),(null as any)),
      i0.ɵqud(335544320,1,{stepLabel:0})],(null as any),(null as any));
}
export const CdkStepNgFactory:i0.ComponentFactory<i1.CdkStep> = i0.ɵccf('cdk-step',
    i1.CdkStep,View_CdkStep_Host_0,{stepControl:'stepControl',label:'label',editable:'editable',
        optional:'optional',completed:'completed'},{},['*']);
