// @angular/material/expansion/typings/index.ngsummary.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/material/expansion';
import * as i1 from '@angular/common';
import * as i2 from '@angular/core';
import * as i3 from '@angular/cdk/platform';
import * as i4 from '@angular/cdk/a11y';
import * as i5 from '@angular/cdk/collections';
import * as i6 from '../../../common/common.ngsummary';
import * as i7 from '../../../cdk/platform/typings/index.ngsummary';
import * as i8 from '../../../cdk/a11y/typings/index.ngsummary';
import * as i9 from './index.ngfactory';
export function MatExpansionModuleNgSummary():any[] {
  return [{summaryKind:2,type:{reference:i0.MatExpansionModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},entryComponents:([] as any[]),providers:[{provider:{token:{identifier:{reference:i1.NgLocalization}},
      useClass:{reference:i1.NgLocaleLocalization,diDeps:[{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
      useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
          isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.LOCALE_ID}}}],
      multi:false},module:{reference:i1.CommonModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i3.Platform,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
          useClass:{reference:i3.Platform,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
          useValue:(undefined as any),useFactory:(null as any),useExisting:(undefined as any),
          deps:([] as any[]),multi:false},module:{reference:i3.PlatformModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i4.InteractivityChecker,
          diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i3.Platform}}}],lifecycleHooks:([] as any[])}},
          useClass:{reference:i4.InteractivityChecker,diDeps:[{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}}],
              lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}}],
          multi:false},module:{reference:i4.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i4.FocusTrapFactory,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i4.InteractivityChecker}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i3.Platform}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}}],
          lifecycleHooks:([] as any[])}},useClass:{reference:i4.FocusTrapFactory,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i4.InteractivityChecker}}},
          {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
              token:{identifier:{reference:i3.Platform}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}}],
          lifecycleHooks:([] as any[])},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i4.InteractivityChecker}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i3.Platform}}},{isAttribute:false,isHost:false,
                  isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}}],
          multi:false},module:{reference:i4.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i4.AriaDescriber,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}}],
          lifecycleHooks:[1]}},useClass:{reference:i4.AriaDescriber,diDeps:[{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}}],
          lifecycleHooks:[1]},useValue:(undefined as any),useFactory:(null as any),
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}}],
          multi:false},module:{reference:i4.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i4.LiveAnnouncer}},useClass:(null as any),
          useValue:(undefined as any),useFactory:{reference:i4.LIVE_ANNOUNCER_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i4.LiveAnnouncer}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,
                      token:{identifier:{reference:i4.LIVE_ANNOUNCER_ELEMENT_TOKEN}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.Platform}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i4.LiveAnnouncer}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i4.LIVE_ANNOUNCER_ELEMENT_TOKEN}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i3.Platform}}}],multi:false},module:{reference:i4.A11yModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i4.AriaDescriber}},
          useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i4.ARIA_DESCRIBER_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i4.AriaDescriber}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.Platform}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i4.AriaDescriber}}},{isAttribute:false,
              isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i3.Platform}}}],
          multi:false},module:{reference:i4.A11yModule,diDeps:([] as any[]),lifecycleHooks:([] as any[])}},
      {provider:{token:{identifier:{reference:i4.FocusMonitor}},useClass:(null as any),
          useValue:(undefined as any),useFactory:{reference:i4.FOCUS_MONITOR_PROVIDER_FACTORY,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i4.FocusMonitor}}},
                  {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i2.NgZone}}},{isAttribute:false,
                      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                      token:{identifier:{reference:i3.Platform}}}]},useExisting:(undefined as any),
          deps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,isOptional:true,
              token:{identifier:{reference:i4.FocusMonitor}}},{isAttribute:false,isHost:false,
              isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.NgZone}}},
              {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
                  token:{identifier:{reference:i3.Platform}}}],multi:false},module:{reference:i4.A11yModule,
          diDeps:([] as any[]),lifecycleHooks:([] as any[])}},{provider:{token:{identifier:{reference:i5.UniqueSelectionDispatcher}},
          useClass:(null as any),useValue:(undefined as any),useFactory:{reference:i5.ɵa,
              diDeps:[{isAttribute:false,isHost:false,isSelf:false,isSkipSelf:true,
                  isOptional:true,token:{identifier:{reference:i5.UniqueSelectionDispatcher}}}]},
          useExisting:(undefined as any),deps:[{isAttribute:false,isHost:false,isSelf:false,
              isSkipSelf:true,isOptional:true,token:{identifier:{reference:i5.UniqueSelectionDispatcher}}}],
          multi:false},module:{reference:i0.MatExpansionModule,diDeps:([] as any[]),
          lifecycleHooks:([] as any[])}}],modules:[{reference:i1.CommonModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},{reference:i3.PlatformModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},{reference:i4.A11yModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},{reference:i0.MatExpansionModule,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])}],exportedDirectives:[{reference:i0.CdkAccordion},
      {reference:i0.MatAccordion},{reference:i0.MatExpansionPanel},{reference:i0.MatExpansionPanelActionRow},
      {reference:i0.MatExpansionPanelHeader},{reference:i0.MatExpansionPanelTitle},
      {reference:i0.MatExpansionPanelDescription}],exportedPipes:([] as any[])},CdkAccordionNgSummary,
      MatAccordionNgSummary,MatExpansionPanelNgSummary,MatExpansionPanelActionRowNgSummary,
      MatExpansionPanelHeaderNgSummary,MatExpansionPanelTitleNgSummary,MatExpansionPanelDescriptionNgSummary,
      i6.CommonModuleNgSummary,i7.PlatformModuleNgSummary,i8.A11yModuleNgSummary];
}
export function CdkAccordionNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.CdkAccordion,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'cdk-accordion, [cdk-accordion]',exportAs:'cdkAccordion',
      inputs:{multi:'multi',hideToggle:'hideToggle',displayMode:'displayMode'},outputs:{},
      hostListeners:{},hostProperties:{},hostAttributes:{},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatAccordionNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatAccordion,diDeps:([] as any[]),lifecycleHooks:([] as any[])},
      isComponent:false,selector:'mat-accordion',exportAs:'matAccordion',inputs:{multi:'multi',
          hideToggle:'hideToggle',displayMode:'displayMode'},outputs:{},hostListeners:{},
      hostProperties:{},hostAttributes:{class:'mat-accordion'},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatExpansionPanelNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatExpansionPanel,diDeps:[{isAttribute:false,
      isHost:true,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.MatAccordion}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ChangeDetectorRef}}},{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.UniqueSelectionDispatcher}}}],
      lifecycleHooks:[1,3]},isComponent:true,selector:'mat-expansion-panel',exportAs:'matExpansionPanel',
      inputs:{disabled:'disabled',expanded:'expanded',hideToggle:'hideToggle'},outputs:{},
      hostListeners:{},hostProperties:{'class.mat-expanded':'expanded','class.mat-expansion-panel-spacing':'_hasSpacing()'},
      hostAttributes:{class:'mat-expansion-panel'},providers:[{token:{identifier:{reference:i0.AccordionItem}},
          useClass:(null as any),useValue:(undefined as any),useFactory:(null as any),
          useExisting:{identifier:{reference:i0.MatExpansionPanel}},deps:(undefined as any),
          multi:false}],viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:['bodyExpansion'],
          ngContentSelectors:['mat-expansion-panel-header','*','mat-action-row'],encapsulation:2},
      componentViewType:i9.View_MatExpansionPanel_0,rendererType:i9.RenderType_MatExpansionPanel,
      componentFactory:i9.MatExpansionPanelNgFactory}];
}
export function MatExpansionPanelActionRowNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatExpansionPanelActionRow,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:false,selector:'mat-action-row',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-action-row'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatExpansionPanelHeaderNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatExpansionPanelHeader,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.Renderer2}}},
      {isAttribute:false,isHost:true,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i0.MatExpansionPanel}}},{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ElementRef}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i4.FocusMonitor}}},{isAttribute:false,isHost:false,
          isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i2.ChangeDetectorRef}}}],
      lifecycleHooks:[1]},isComponent:true,selector:'mat-expansion-panel-header',exportAs:(null as any),
      inputs:{expandedHeight:'expandedHeight',collapsedHeight:'collapsedHeight'},outputs:{},
      hostListeners:{click:'_toggle()',keyup:'_keyup($event)'},hostProperties:{'attr.tabindex':'panel.disabled ? -1 : 0',
          'attr.aria-controls':'_getPanelId()','attr.aria-expanded':'_isExpanded()',
          'attr.aria-disabled':'panel.disabled','class.mat-expanded':'_isExpanded()',
          '@expansionHeight':'{\n        value: _getExpandedState(),\n        params: {\n          collapsedHeight: collapsedHeight,\n          expandedHeight: expandedHeight\n        }\n    }'},
      hostAttributes:{class:'mat-expansion-panel-header',role:'button'},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:0,template:{animations:['indicatorRotate',
          'expansionHeight'],ngContentSelectors:['mat-panel-title','mat-panel-description',
          '*'],encapsulation:2},componentViewType:i9.View_MatExpansionPanelHeader_0,
      rendererType:i9.RenderType_MatExpansionPanelHeader,componentFactory:i9.MatExpansionPanelHeaderNgFactory}];
}
export function MatExpansionPanelTitleNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatExpansionPanelTitle,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:false,selector:'mat-panel-title',exportAs:(null as any),
      inputs:{},outputs:{},hostListeners:{},hostProperties:{},hostAttributes:{class:'mat-expansion-panel-header-title'},
      providers:([] as any[]),viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function MatExpansionPanelDescriptionNgSummary():any[] {
  return [{summaryKind:1,type:{reference:i0.MatExpansionPanelDescription,diDeps:([] as any[]),
      lifecycleHooks:([] as any[])},isComponent:false,selector:'mat-panel-description',
      exportAs:(null as any),inputs:{},outputs:{},hostListeners:{},hostProperties:{},
      hostAttributes:{class:'mat-expansion-panel-header-description'},providers:([] as any[]),
      viewProviders:([] as any[]),queries:([] as any[]),viewQueries:([] as any[]),
      entryComponents:([] as any[]),changeDetection:(null as any),template:(null as any),
      componentViewType:(null as any),rendererType:(null as any),componentFactory:(null as any)}];
}
export function AccordionItemNgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.AccordionItem,diDeps:[{isAttribute:false,
      isHost:false,isSelf:false,isSkipSelf:false,isOptional:true,token:{identifier:{reference:i0.CdkAccordion}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ChangeDetectorRef}}},{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.UniqueSelectionDispatcher}}}],
      lifecycleHooks:[1]}}];
}
export function ɵa15NgSummary():any[] {
  return [{summaryKind:3,type:{reference:i0.ɵa15,diDeps:[{isAttribute:false,isHost:false,
      isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i0.MatAccordion}}},
      {isAttribute:false,isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,
          token:{identifier:{reference:i2.ChangeDetectorRef}}},{isAttribute:false,
          isHost:false,isSelf:false,isSkipSelf:false,isOptional:false,token:{identifier:{reference:i5.UniqueSelectionDispatcher}}}],
      lifecycleHooks:[1]}}];
}
